### PR TITLE
Complete the Transaction Processor for the rust sdk

### PR DESCRIPTION
**Pull request by agunde406**
This pr includes adding a monitor socket and handling a
disconnection from the validator, and ability to respond to
ping requests. Also the handling of wrong message types of
has been updated to match the other SDK and unwraps have
been removed in the processor.

Also includes some other random fixups.

### CODE REVIEW

File: sdk/rust/src/processor/mod.rs [hunk 1]
```diff
@@ -225,6 +226,34 @@ impl<'a> TransactionProcessor<'a> {
...
+                                let response = PingResponse::new();
+                                let serialized = match response.write_to_bytes() {
+                                    Ok(serialized) => serialized,
+                                    Err(err) => {
+                                        error!("Serialization failed: {}", err.description());
+                                        continue
+                                    }
+                                };
+                                let x : &[u8] = &serialized;
+                                match sender.reply(
```
Comment (peterschwarz) [inline sdk/rust/src/processor/mod.rs]: Have we determined if we can use `rustfmt` yet?  Not wanting to ding this line based on formatting without a tool, but I'm registering my opinion that I'm not a fan of the way this line is wrapped.

Comment (agunde406) [reply sdk/rust/src/processor/mod.rs]: Opinion registered. I do not believe that we can run rustfmt yet but that is something I can look into shortly.

File: sdk/rust/Cargo.toml [hunk 2]
```diff
@@ -24,7 +24,8 @@ protobuf="1.4.1"
 secp256k1 = "0.7.1"
 rand = "0.4.2"
 rust-crypto = "0.2.36"
-zmq = "0.8"
+rust-base58 = "0.0.4"
```
Comment (aludvik) [inline sdk/rust/Cargo.toml]: We can't use this library due to licensing

File: sdk/rust/src/messaging/zmq_stream.rs [hunk 3]
```diff
@@ -171,14 +175,25 @@ impl InboundRouter {
...
+    fn route(&mut self, message_result: MessageResult) {
+        match message_result  {
+            Ok(message) => {
+                let mut expected_replies = self.expected_replies.lock().unwrap();
+                match expected_replies.remove(
+                message.get_correlation_id())
+                {
+                Some(sender) => sender.send(Ok(message)).unwrap(),
+                None => self.inbound_tx.send(Ok(message)).unwrap()
+                }
```
Comment (aludvik) [inline sdk/rust/src/messaging/zmq_stream.rs]: Above indentation is weird

**Pull request merged**